lib/link_checker.rb

```diff
@@ -33,4 +35,25 @@ def get_urls_from_locale_files
     end
     [gov_urls, other_urls]
   end
+
+  def find_invalid_govuk_paths govuk_urls
+    publishing_api = GdsApi::PublishingApi.new(
+      "https://publishing-api.publishing.service.gov.uk",
+      bearer_token: ENV["PUBLISHING_API_BEARER_TOKEN"],
+    )
+    govuk_paths = govuk_urls.map { |url| URI(url).path }
+
+    puts publishing_api
+    unpublished_paths = publishing_api.lookup_content_ids(
```

Comment:
I think this variable should be renamed to `all_base_paths` as it's not just getting unpublished base paths, it's getting all of them. 
Also, unpublished is a bit of an overloaded term on GOV.UK as it can refer to content items that have been withdrawn, completely removed or redirected, but doesn't include content items that haven't been published yet (drafts).

Reply:
It's probably possible to get the current state of the content item, but I think you'd need to call `get_content` on each with the `content_id`, and from that be able to get the state.